### PR TITLE
Add a new workflow to trigger tests for a community contribution (from a fork)

### PR DESCRIPTION
**Pull request by idodod**
This PR would add new target and GHA to open a PR based on a PR coming from a fork, once the original PR is applied with a label (See the added document).

Also:
 - Enabling Renovate for GHA files (last attempt was not done properly)
 - Enabling Renovate for the gh cli which is used in the new target

### CODE REVIEW

File: docs-internals/contributions.md
Comment (alexcb) [inline docs-internals/contributions.md:1]: can we put this under the existing root CONTRIBUTING.md instead?

File: docs-internals/contributions.md
Comment (alexcb) [inline docs-internals/contributions.md:8]: > This would trigger a GHA that would open a new draft PR

I'm a bit confused on how this can work -- doesn't this require access to our secrets?

File: docs-internals/contributions.md
Comment (idodod) [reply docs-internals/contributions.md:1]: I thought about it, but it seems to be like `CONTRIBUTING.md` is more directed to the community whilst what I'm trying to capture here is for us internally.

File: Earthfile
Comment (alexcb) [inline Earthfile:902]: you might also need to have a 

```
git commit --allow-empty -m "please run the test"
```

for cases where no merge is required.

this is needed because github actions won't re-run the job if a previous job has already been started on the same commit.

**Comment by idodod** [comment]
Hey @alexcb , should we use this method until we come up with a better alternative?
As a reminder, this is the "fork the community PR branch and open a new PR which is triggered after adding a `approved-for-tests`" method.

**Review by alexcb — APPROVED** [review]
looks good!

I wonder if we could have it automatically add a comment to outside-contributor PRs, along the lines of "thanks for the PR; tests will be automatically run once an earthly maintainer adds a `approved-for-tests` label; see the CONTRIBUTING guide for details".

**Comment by idodod** [comment]
> looks good!
> 
> I wonder if we could have it automatically add a comment to outside-contributor PRs, along the lines of "thanks for the PR; tests will be automatically run once an earthly maintainer adds a `approved-for-tests` label; see the CONTRIBUTING guide for details".

sure, I need to clean up this pr a little bit anyway, I can add a workflow to auto reply as you suggested.

**Pull request merged**
